pypsa/plot.py

```diff
@@ -1233,3 +1243,160 @@ def iplot(
+            LineString(
+                [
+                    (n.buses.loc[line.bus0, "x"], n.buses.loc[line.bus0, "y"]),
+                    (n.buses.loc[line.bus1, "x"], n.buses.loc[line.bus1, "y"]),
+                ]
+            )
+            for line in n.lines.itertuples()
+        ],
+        crs=crs,
+    )
+
+    gdf_transformers = gpd.GeoDataFrame(
+        n.transformers,
+        geometry=[
+            LineString(
+                [
+                    (
+                        n.buses.loc[transformer.bus0, "x"],
+                        n.buses.loc[transformer.bus0, "y"],
+                    ),
+                    (
+                        n.buses.loc[transformer.bus1, "x"],
+                        n.buses.loc[transformer.bus1, "y"],
+                    ),
+                ]
+            )
+            for transformer in n.transformers.itertuples()
+        ],
+        crs=crs,
+    )
+
+    # Map related settings
+    bus_colors = mcolors.CSS4_COLORS["cadetblue"]
+    line_colors = mcolors.CSS4_COLORS["rosybrown"]
+    link_colors = mcolors.CSS4_COLORS["darkseagreen"]
+    transformer_colors = mcolors.CSS4_COLORS["orange"]
+
+    # Initialize the map
+    map = Map(tiles=None)
+
+    # Add tile layer legend entries
+    TileLayer("CartoDB dark_matter", name="CartoDB Dark Matter").add_to(map)
+    TileLayer("CartoDB positron", name="CartoDB Positron").add_to(map)
+    TileLayer("OpenStreetMap", name="OpenStreetMap").add_to(map)
+
+    components = ["buses", "lines", "links", "transformers"]
+    components_present = []
+
+    if not gdf_transformers.empty:
+        gdf_transformers.explore(
+            m=map,
+            color=transformer_colors,
+            tooltip=tooltip,
+            popup=popup,
+            name="Transformers",
+        )
+        components_present.append("transformers")
+
+    if not gdf_lines.empty:
+        gdf_lines.explore(
+            m=map, color=line_colors, tooltip=tooltip, popup=popup, name="Lines"
+        )
+        components_present.append("lines")
+
+    if not gdf_links.empty:
+        gdf_links.explore(
+            m=map, color=link_colors, tooltip=tooltip, popup=popup, name="Links"
+        )
+        components_present.append("links")
+
+    if not gdf_buses.empty:
+        buses_group = FeatureGroup(name="Buses")
+        gdf_buses.explore(
+            m=map,
+            color=bus_colors,
+            tooltip=tooltip,
+            popup=popup,
+            name="Buses",
+            marker_kwds={"radius": 3},
+        )
+        components_present.append("buses")
+
+    logger_explore = ""
+    if len(components_present) > 0:
+        logger_explore += (
+            f"Components rendered on the map: {', '.join(components_present)}."
+        )
+    if len(set(components) - set(components_present)) > 0:
+        logger_explore += f" Components omitted as they are missing: {', '.join(set(components) - set(components_present))}."
+    logger.info(logger_explore)
```

Comment:
Just use logger.info directly here. You don't need logger_explore as an interim step

Reply:
great, thanks!